mne/viz/backends/_notebook.py

```diff
@@ -10,58 +10,201 @@
 
 class _Renderer(_PyVistaRenderer):
```

Comment:
This inhertance is a bit weird, because it overrides a bunch of dock-related stuff from `_PyVistaRenderer`. How about:

1. `_PyVistaRenderer`, which handles rendering using PyVista. It is a subclass of `_AbstractRenderer` and `_AbstractDock`.
2. `_QtRenderer`, which handles dock-related stuff, subclasses both `_PyVistaRenderer` and `_QtDock`, and lives in a new `_qt.py` file (with `_QtDock`).
3. `_IPyRenderer`, which handles ipywidgets dock-related stuff, subclasses both `_PyVistaRenderer` and `_IPyDock`, the latter providing the dock methods, and lives in this `_notebook.py` file (with _IPyDock).

And really I'm not sure if we even need separate `_QtDock` and `_IPyDock` classes, these could just implement the dock methods directly on `_QtRenderer` and `_IPyRenderer`. Up to you...

Either way, this has the advantage that all Qt imports can be done at the top of `_qt.py`, and all ipywidgets can be done at the top of `_notebook.py`, etc. But more importantly, it becomes more readable and more maintainable because the hierarchy is clearer -- the split in files and classes reflects the split in functionality.

Reply:
What  do you think of ?

* In `pyvista`:
  `_PyVistaRenderer` <- `_AbstractRenderer`
   `_Renderer` <- `_PyVistaRenderer`, `_AbstractDock`, `_AbstractToolBar` 
   
 * In `notebook`:
  `_Renderer` <- `_PyVistaRenderer`, `_AbstractDock`, `_AbstractToolBar`

Reply:
I followed it in 4228bd263e3f306c8d4d2e15cdedb0c1fa2e4520

Reply:
Oh I forgot to split in different files though

Reply:
With different files to separate implementations:

* In `pyvista`:
   `_PyVistaRenderer` <- `_AbstractRenderer`
   `_QtDock` <- `_AbstractDock` (actual implementation)
   `_QtToolBar` <- `_AbstractToolBar` (actual implementation)
   `_Renderer` <- `_PyVistaRenderer`, `_QtDock`, `_QtToolBar` 
   
 * In `notebook`:
    `_IpyDock` <- `_AbstractDock` (actual implementation)
    `_IpyToolBar` <- `_AbstractToolBar` (actual implementation)
    `_Renderer` <- `_PyVistaRenderer`, `_IpyDock`, `_IpyToolBar`